src/@providers/s3/services/secret.ts

```diff
@@ -0,0 +1,135 @@
+import { Subscriber } from 'rxjs';
+import { ResourceInputs, ResourceOutputs } from '../../../@resources/index.ts';
+import { PagingOptions, PagingResponse } from '../../../utils/paging.ts';
+import { DeepPartial } from '../../../utils/types.ts';
+import { CrudResourceService } from '../../crud.service.ts';
+import { S3Credentials } from '../credentials.ts';
+import S3Utils from '../utils.ts';
+
+const ARCCTL_SECRET_TYPE = 'arcctl-secret';
+
+export class S3SecretService extends CrudResourceService<'secret', S3Credentials> {
+  async get(id: string): Promise<ResourceOutputs['secret'] | undefined> {
+    const client = S3Utils.getS3Client(this.credentials);
+    const idParts = id.split('--');
+    if (idParts.length < 2) {
+      throw new Error(`The secret name for ${id} is invalid`);
+    }
+    const namespace = idParts.shift();
+    const name = idParts.join('--');
+    const object = await client.getObject({
+      Bucket: namespace!,
+      Key: name,
+    }).promise();
+    if (!object.Body) {
+      throw new Error(`The ${id} secret does not exist`);
+    }
+
+    return Promise.resolve({
+      id: id,
+      data: object.Body.toString(),
+    });
+  }
+
+  async list(
+    _filterOptions?: Partial<ResourceOutputs['secret']> | undefined,
+    _pagingOptions?: Partial<PagingOptions> | undefined,
+  ): Promise<PagingResponse<ResourceOutputs['secret']>> {
+    const client = S3Utils.getS3Client(this.credentials);
+
+    const secrets: ResourceOutputs['secret'][] = [];
+
+    const buckets = await client.listBuckets().promise();
+    for (const bucket of buckets.Buckets || []) {
+      const objects = await client.listObjects({
+        Bucket: bucket.Name!,
+      }).promise();
+      for (const object of objects.Contents || []) {
+        if (object.Key) {
+          const objectMetadata = await client.headObject({
+            Bucket: bucket.Name!,
+            Key: object.Key,
+          }).promise();
+          if (objectMetadata.Metadata?.type !== ARCCTL_SECRET_TYPE) {
```

Comment:
It doesn't feel right that arcctl specific references like this would live inside a provider. That almost implies that ALL providers that support secrets would need to do something similar?

Reply:
How else do we know if an object in the bucket is a secret or just some object? Do we auto assume all files in a bucket are a secret? 
Also not all other providers would need to do this, it really is up to the provider. if we used vault for instance then its easy to determine what is a secret vs what is not a secret. This is more due to the fact that S3 is a generic storage system that happens to be used for secrets.

Reply:
This is literally a service for the resource type `secret`. I would definitely assume all files are "secrets".